### PR TITLE
Fix crash after entering manage content

### PR DESCRIPTION
**Pull request by teinarss**
Fixes a crash that happens after entering Manage content screen. The problem was that `static readonly Lazy<DiscordService> Service = Exts.Lazy(() =>` held a reference to the disposed discord client.

### CODE REVIEW

File: OpenRA.Mods.Common/DiscordService.cs [hunk 1]
```diff
@@ -32,16 +32,21 @@ public enum DiscordState
...
-		static readonly Lazy<DiscordService> Service = Exts.Lazy(() =>
+		static DiscordService instance;
+		static DiscordService GetService()
 		{
 			if (!Game.ModData.Manifest.Contains<DiscordService>())
 				return null;
 
-			return Game.ModData.Manifest.Get<DiscordService>();
-		});
+			if (instance != null)
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/DiscordService.cs:41]: This can/should move above the `Contains` check.

**Comment by teinarss** [comment]
Updated.

**Review by pchote — APPROVED** [review]

**Comment by abcdefg30** [comment]
Fwiw I can still reproduce #18508.

**Review by abcdefg30 — APPROVED** [review]
Content crash is fixed though.

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/ffbf30f1775cc0796185234453e1be79d4353c83...9c44897856131ff145299a52f2accd1df67d9724)

**Pull request merged**